code-challenge/src/App.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {useEffect, useState} from 'react';
+import './App.css';
+import ApiService from './services/ApiService';
+import { AppBar, Toolbar, Typography } from "@material-ui/core";
+import ArticleList from "./components/ArticleList/ArticleList.lazy";
+import Search from "./components/Search/Search.lazy";
+import { useDispatch, useSelector } from 'react-redux';
+import { store } from "./redux/createStore";
+
+type RootState = ReturnType<typeof store.getState>;
+
+
+function App() {
+  const [error, setError] = useState(null);
+  const searchTerm = useSelector((state: RootState) => state.searchTerm);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const apiService = new ApiService();
```

Comment:
I'm curious of your thoughts of Services being a collection of functions vs a class that needs to be instantiated?

Reply:
I like the idea of classes when there are multiple things that may be shared. For example, in an API service, adding things like content-type headers and authentication headers can be useful. They can also share error handlers since HTTP codes can be sent for any type of request. If the "things" are related but perhaps don't need to share much logic like this, I think the collection makes more sense. That said, another approach would be to curry functions for that shared logic, but my object-oriented brain throws alerts.